src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java

```diff
@@ -59,23 +62,40 @@ protected Object decode(
             return null;
         }
 
-        Position position = new Position(getProtocolName());
-
         DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, parser.next());
         if (deviceSession == null) {
             return null;
         }
+        
+        Position position = new Position(getProtocolName());
         position.setDeviceId(deviceSession.getDeviceId());
-
+        long status = parser.nextHexLong();
+        String model = parser.next();
         position.setValid(parser.next().equals("A"));
         position.setLatitude(parser.nextCoordinate());
         position.setLongitude(parser.nextCoordinate());
-        position.setSpeed(parser.nextDouble(0));
+        position.setSpeed(UnitsConverter.kphFromKnots(parser.nextDouble(0)));
         position.setCourse(parser.nextDouble(0));
-
         position.set(Position.KEY_HDOP, parser.nextDouble(0));
-
         position.setTime(parser.nextDateTime(Parser.DateTimeFormat.HMS_DMY));
+        if (model.contains("PT33")) {
```

Comment:
Why is this not using `equals` instead of `PT33`

Reply:
changed to equals